Restructure repo so that existing tests are separate and findable by pytest

jatkinson1000 opened the pull request:
There is some existing testing code in the project. We make these compatible with pytest and tweak a test for the refactored CNN module introduced in #20 .

The CI runs `pytest` correctly as of writing.

Original issue text below.

---

The whole repository is in need of tests.
Use pytest.

As a starting point:
- [ ] `data/`
  - [ ] ?
- [ ] `model/`
  - [ ] ?

- [ ] Add to CI

These routines should then be merged into other branches and run before further PRs are closed.

Closes #6

dorchard commented:
- [ ] Move this test in `models1.py` https://github.com/m2lines/GZ21_ocean_momentum_CNN/blob/main/models/models1.py#L155-L161 into its own test file.

raehik commented:
First task would be to move `data/test_coarse.py` tests into a pytest module (what is the standard practice there?)

jatkinson1000 commented:
I moved `data/test_course.py` to `data/tests/test_course.py` when opening this PR.
This is the standard format to be picked up by pytest.

I also rewrote it as a pytest test instead of unittest on the old branch [here](https://github.com/m2lines/GZ21_ocean_momentum_CNN/blob/cleanup_jwa/src/gz_ocean_momentum/data/test_coarse.py) which may help.

dorchard commented:
Refined the scope of this PR a bit more so that it just handles the existing tests, putting them in a more regular form that can be run by pytest.

mondus commented on pyproject.toml:
```diff
@@ -30,6 +30,8 @@ dependencies = [
     "intake-xarray",
     "gcsfs",
 
+    "matplotlib", # analysis, examples
```
Specify the minimum version

mondus commented on pyproject.toml:
```diff
@@ -30,6 +30,8 @@ dependencies = [
+    "matplotlib", # analysis, examples
+
     "torch>=1.13.1",
     "progressbar>=2.5" # testing/utils (but imported by trainScript)
```
Add `cartopy` as a dependency as this is used in Analysis.

mondus reviewed (changes requested):
Arthur is happy with the removal of test_anim.py. We have approved the movement of models1.py to the `test` directory in the file `test_fully_conv_net`.
